src/Evolution/Systems/ScalarAdvection/BoundaryCorrections/Rusanov.cpp

```diff
@@ -0,0 +1,96 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/ScalarAdvection/BoundaryCorrections/Rusanov.hpp"
+
+#include <memory>
+#include <optional>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Formulation.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace ScalarAdvection::BoundaryCorrections {
+template <size_t Dim>
+Rusanov<Dim>::Rusanov(CkMigrateMessage* msg) noexcept
+    : BoundaryCorrection<Dim>(msg) {}
+
+template <size_t Dim>
+std::unique_ptr<BoundaryCorrection<Dim>> Rusanov<Dim>::get_clone()
+    const noexcept {
+  return std::make_unique<Rusanov>(*this);
+}
+
+template <size_t Dim>
+void Rusanov<Dim>::pup(PUP::er& p) {
+  BoundaryCorrection<Dim>::pup(p);
+}
+
+template <size_t Dim>
+double Rusanov<Dim>::dg_package_data(
+    const gsl::not_null<Scalar<DataVector>*> packaged_u,
+    const gsl::not_null<Scalar<DataVector>*> packaged_normal_dot_flux_u,
+    const gsl::not_null<Scalar<DataVector>*> packaged_abs_char_speed,
+    const Scalar<DataVector>& u,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& flux_u,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& velocity_field,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_covector,
+    const std::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+    /*mesh_velocity*/,
+    const std::optional<Scalar<DataVector>>&
+        normal_dot_mesh_velocity) noexcept {
+  Scalar<DataVector>& normal_dot_velocity = *packaged_u;
+  dot_product(make_not_null(&normal_dot_velocity), velocity_field,
+              normal_covector);
+  if (normal_dot_mesh_velocity.has_value()) {
+    get(*packaged_abs_char_speed) =
+        abs(get(normal_dot_velocity) - get(*normal_dot_mesh_velocity));
+  } else {
+    get(*packaged_abs_char_speed) = abs(get(normal_dot_velocity));
+  }
+  *packaged_u = u;
+  dot_product(packaged_normal_dot_flux_u, flux_u, normal_covector);
```

Comment:
Here, please use `normal_dot_flux(packaged_normal_dot_flux_u, normal_covector, flux_u);` and include the file `NumericalAlgorithms/DiscontinuousGalerkin/NormalDotFlux.hpp`. The current fluxes don't do that very consistently, but I am putting in a PR soon that will fix that.

Reply:
Okay. Can I ask you what a notable difference between `normal_dot_flux` and `dot_product` is in this context?

Reply:
There's no notable difference here because `u` is a Scalar. But `normal_dot_flux` will handle all the for-loops if the evolved variable is a vector or a tensor... for example in the hydro case one of the evolved variables is a momentum with components u_x, u_y, u_z, and instead of writing three different `dot_product` calls we can use a single `normal_dot_flux` call. Does that make sense? My PR will use `normal_dot_flux` in all the other bdry corrections.

Reply:
Got it. Thank you!